shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt

```diff
@@ -285,3 +295,22 @@ fun String.extractShortTypeFromExpanded(): String =
      * TODO do a clean implementation using info from @context
      */
     this.substringAfterLast("/").substringAfterLast("#")
+
+fun Map<String, Any>.toKeyValues(): Map<String, Any> {
+    return this.map { (key, value) -> key to simplifyValueIfNeeded(value) }.toMap()
+}
+
+private fun simplifyValueIfNeeded(value: Any): Any {
+    return when (value) {
+        is Map<*, *> -> simplifyValue(value)
+        else -> value
```

Comment:
Does it happen? When?

Reply:
when the value is not a map of course

Reply:
Yes, I can read it. So ... when the value is not a map (which use case? for which value? ...)?

Reply:
id, type, @context

Reply:
as discussed, a one line comment to give some context would be great

Reply:
ok